### PR TITLE
fix install kind argument invalid

### PR DESCRIPTION
**Pull request by seanly**
1. fix kind version argument invalid
2. fix ksVersion argument invalid
3. fix kind cluster name argument invalid

test result:
image

### CODE REVIEW

**Review by LinuxSuRen — COMMENTED** [review]
Thanks for helping fix some issues. But I didn't quite understand what the issues are. Can you just make them be clear?

**Comment by seanly** [comment]
> Thanks for helping fix some issues. But I didn't quite understand what the issues are. Can you just make them be clear?

In the current version, the following use cases are exceptions

test case1: 
> execute command: 
bin/ks install kind --version=v1.19.7 --name=ks
expect:
kind version v1.19.7
actual:
kind version v1.18.2

test case2:
> execute command: 
bin/ks install kind --version=v1.19.7 --name=ks
expect: 
the image load uses kind cluster name is `ks`
actual:
the image load uses kind cluster name is `kind`

test case3:
> execute command: 
bin/ks install kind --version=v1.19.7 --name=ks
expect: 
ks-installer version is `v3.1.1`
actual:
Is still the image of the installed v3.0.0 without using the image specified by ksVersion

File: kubectl-plugin/install/kind.go [hunk 1]
```diff
@@ -86,17 +89,27 @@ func (o *kindOption) preRunE(_ *cobra.Command, _ []string) (err error) {
 }
 
 func (o *kindOption) runE(cmd *cobra.Command, args []string) (err error) {
-	writeConfigFile("config.yaml", o.portMappings)
+
```
Comment (LinuxSuRen) [inline kubectl-plugin/install/kind.go:92]: why left an empty line here?

File: kubectl-plugin/install/kind.go [hunk 2]
```diff
@@ -86,17 +89,27 @@ func (o *kindOption) preRunE(_ *cobra.Command, _ []string) (err error) {
...
+	kindImage := fmt.Sprintf("kindest/node:%s", o.version)
+
+	kindConfig := KindConfig{
+		Image:        kindImage,
+		PortMappings: o.portMappings,
+	}
+
+	kindConfigF := path.Join(os.TempDir(), "config.yaml")
```
Comment (LinuxSuRen) [inline kubectl-plugin/install/kind.go:100]: I'd suggest deleting this temporary file after this command was done.

File: kubectl-plugin/install/kind.go [hunk 3]
```diff
@@ -123,94 +136,79 @@ func (o *kindOption) runE(cmd *cobra.Command, args []string) (err error) {
...
 }
 
-func pullAndLoadImageSync(image string, wg *sync.WaitGroup) (err error) {
+func pullAndLoadImageSync(image string, kindName string, wg *sync.WaitGroup) (err error) {
 	wg.Add(1)
 	go func(imageName string, wgInner *sync.WaitGroup) {
-		_ = pullAndLoadImage(imageName)
+		commander := Commander{}
+		if err = commander.execCommand("docker", "pull", image); err == nil {
+			err = commander.execCommand("kind", "load", "docker-image", "--name", kindName, image)
```
Comment (LinuxSuRen) [inline kubectl-plugin/install/kind.go:190]: Please don't use `kindName` directly. You can add a new variable in line 187.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: kubectl-plugin/install/kind.go (hunk 1 above)
Comment (seanly) [reply kubectl-plugin/install/kind.go:92]: OK, I'll fix it.

**Comment by LinuxSuRen** [comment]
hi @seanly , I don't recommend you force push the commits. I need to review all changed files because I cannot figure out what changes do you make in your last commit.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Comment by seanly** [comment]
> hi @seanly , I don't recommend you force push the commits. I need to review all changed files because I cannot figure out what changes do you make in your last commit.

**Comment by LinuxSuRen** [comment]
>  I combined the two similar changes recently, Does not affect review already.

It's not true. The maintainer must review all changes. No one can guarantee for the changes.

File: kubectl-plugin/install/kind.go [hunk 4]
```diff
@@ -123,106 +146,100 @@ func (o *kindOption) runE(cmd *cobra.Command, args []string) (err error) {
...
-		return
-	}
-	if err = pullAndLoadImageSync("kubesphere/builder-nodejs:v2.1.0", &wg); err != nil {
-		return
-	}
-	if err = pullAndLoadImageSync("kubesphere/builder-go:v2.1.0", &wg); err != nil {
-		return
+
+	images := map[string]string{
+		"kubesphere/jenkins-uc":     "v3.0.0",
```
Comment (JohnNiang) [inline kubectl-plugin/install/kind.go:153]: IMO, some of them are deprecated and should not be used. Or we can determine the tags according to version of KubeSphere.

Comment (JohnNiang) [reply kubectl-plugin/install/kind.go:153]: But the issue is unrelated with the PR. So this won't stop another maintainers from merging this though.

**Pull request merged**